In [1]:
import pandas as pd
import numpy as np

In [2]:
NASdata = pd.read_csv("C:/Users/Rohith007/Downloads/ISB_CBA/BigDataManagement 1/71710097_BDM_Assignment_1/nas-pupil-marks.csv")

In [3]:
# Create a new column called df.Is_South_India where the value is 1
# if it is a South Indian State, else 0.
NASdata['Is_South_India'] = np.where((NASdata['State']=='AN')|(NASdata['State']=='AP')|(NASdata['State']=='GA')|(NASdata['State']=='KA')|(NASdata['State']=='KL')|(NASdata['State']=='PY')|(NASdata['State']=='TN'), 1, 0)

In [4]:
NASdata_Maths_Science = NASdata.drop(['STUID','Reading %','Social %'], axis = 1)
NASdata_Maths_Science['Mean'] = NASdata_Maths_Science[['Maths %','Science %']].mean(axis=1)
NASdata_Maths_Science.head()

State  District  Gender  Age  Category  Same language  Siblings  Handicap  \
0    AP         1       1    3         3              1         5         2   
1    AP         1       2    3         4              2         5         2   
2    AP         1       2    3         4              2         5         2   
3    AP         1       2    3         3              2         4         2   
4    AP         1       2    3         3              2         5         2   

   Father edu  Mother edu   ...    Express science views  Watch TV  \
0           1           1   ...                        3         3   
1           2           1   ...                        3         4   
2           1           1   ...                        3         4   
3           1           1   ...                        3         4   
4           1           1   ...                        3         2   

   Read magazine  Read a book Play games  Help in household  Maths %  \
0              4            3          4                  4    20.37   
1              4            3          4                  4    12.96   
2              3            3          4                  4    27.78   
3              3            3          4                  4      NaN   
4              3            3          4                  4      NaN   

   Science %  Is_South_India    Mean  
0      27.78               1  24.075  
1      38.18               1  25.570  
2        NaN               1  27.780  
3        NaN               1     NaN  
4      14.55               1  14.550  

[5 rows x 63 columns]

In [5]:
NASdata_Maths_Science_New = NASdata_Maths_Science.drop(['Maths %','Science %'], axis = 1)
NASdata_Maths_Science_Mean = NASdata_Maths_Science_New.dropna()
NASdata_Maths_Science_Mean.head()

State  District  Gender  Age  Category  Same language  Siblings  Handicap  \
0    AP         1       1    3         3              1         5         2   
1    AP         1       2    3         4              2         5         2   
2    AP         1       2    3         4              2         5         2   
4    AP         1       2    3         3              2         5         2   
6    AP         1       2    3         4              2         2         2   

   Father edu  Mother edu   ...    Conduct experiments  \
0           1           1   ...                      3   
1           2           1   ...                      3   
2           1           1   ...                      3   
4           1           1   ...                      3   
6           1           1   ...                      3   

   Solve science problems  Express science views  Watch TV Read magazine  \
0                       3                      3         3             4   
1                       3                      3         4             4   
2                       3                      3         4             3   
4                       3                      3         2             3   
6                       3                      3         4             4   

   Read a book  Play games  Help in household  Is_South_India    Mean  
0            3           4                  4               1  24.075  
1            3           4                  4               1  25.570  
2            3           4                  4               1  27.780  
4            3           4                  4               1  14.550  
6            4           4                  4               1  40.740  

[5 rows x 61 columns]

In [6]:
NASdata_Maths_Science_Mean.shape

(148652, 61)

In [7]:
NASdata_Maths_Science_Mean.columns.values

array(['State', 'District', 'Gender', 'Age', 'Category', 'Same language',
       'Siblings', 'Handicap', 'Father edu', 'Mother edu',
       'Father occupation', 'Mother occupation', 'Below poverty',
       'Use calculator', 'Use computer', 'Use Internet', 'Use dictionary',
       'Read other books', '# Books', 'Distance', 'Computer use',
       'Library use', 'Like school', 'Subjects', 'Give Lang HW',
       'Give Math HW', 'Give Scie HW', 'Give SoSc HW', 'Correct Lang HW',
       'Correct Math HW', 'Correct Scie HW', 'Correct SocS HW',
       'Help in Study', 'Private tuition', 'English is difficult',
       'Read English', 'Dictionary to learn', 'Answer English WB',
       'Answer English aloud', 'Maths is difficult', 'Solve Maths',
       'Solve Maths in groups', 'Draw geometry', 'Explain answers',
       'SocSci is difficult', 'Historical excursions',
       'Participate in SocSci', 'Small groups in SocSci',
       'Express SocSci views', 'Science is difficult',
       'Observe exp

In [8]:
# import preprocessing from sklearn
from sklearn import preprocessing

# encode labels with numeric values between 0 and n_classes-1.
le = preprocessing.LabelEncoder()

In [9]:
# FIT AND TRANSFORM for Mean Dataset
# use df.apply() to apply le.fit_transform to all columns
NASdata_Mean_New = NASdata_Maths_Science_Mean
NASdata_Mean_Encoded = NASdata_Mean_New.apply(le.fit_transform)

#Drop remaining Columns
NASdata_Mean_Encoded_X = NASdata_Mean_Encoded.drop(['District', 'Gender', 'Age', 'Category', 'Same language', 'Siblings', 'Handicap', 'Father edu', 'Mother edu','Father occupation', 'Mother occupation', 'Below poverty','Use calculator', 'Use Internet', 'Use dictionary','Read other books', '# Books', 'Distance', 'Computer use','Library use', 'Like school', 'Give Lang HW','Give Math HW', 'Give Scie HW', 'Give SoSc HW', 'Correct Lang HW','Correct Math HW', 'Correct Scie HW', 'Correct SocS HW', 'Help in Study', 'Private tuition', 'English is difficult', 'Read English', 'Dictionary to learn', 'Answer English WB', 'Answer English aloud', 'Maths is difficult', 'Solve Maths','Solve Maths in groups', 'Draw geometry', 'Explain answers','SocSci is difficult', 'Historical excursions','Participate in SocSci', 'Small groups in SocSci','Express SocSci views', 'Science is difficult','Observe experiments', 'Conduct experiments', 'Solve science problems', 'Express science views', 'Watch TV', 'Read magazine', 'Read a book', 'Play games', 'Help in household','Is_South_India','Mean'], axis = 1)

#Drop the 3 columns to include the same encoded columns
NASdata_Mean_New = NASdata_Mean_New.drop(['State','Use computer','Subjects'], axis = 1)
# Concatinating the DataFrames
NASdata_Mean_New1 = pd.concat([NASdata_Mean_Encoded_X, NASdata_Mean_New], axis=1)

#Rearrange the columns in a proper order
NASdata_Mean_New1 =  pd.DataFrame(NASdata_Mean_New1, columns=['State','District', 'Gender', 'Age', 'Category', 'Same language', 'Siblings', 'Handicap', 'Father edu', 'Mother edu','Father occupation', 'Mother occupation', 'Below poverty','Use calculator','Use computer', 'Use Internet', 'Use dictionary','Read other books', '# Books', 'Distance', 'Computer use','Library use', 'Like school','Subjects', 'Give Lang HW','Give Math HW', 'Give Scie HW', 'Give SoSc HW', 'Correct Lang HW','Correct Math HW', 'Correct Scie HW', 'Correct SocS HW', 'Help in Study', 'Private tuition', 'English is difficult', 'Read English', 'Dictionary to learn', 'Answer English WB', 'Answer English aloud', 'Maths is difficult', 'Solve Maths','Solve Maths in groups', 'Draw geometry', 'Explain answers','SocSci is difficult', 'Historical excursions','Participate in SocSci', 'Small groups in SocSci','Express SocSci views', 'Science is difficult','Observe experiments', 'Conduct experiments', 'Solve science problems', 'Express science views', 'Watch TV', 'Read magazine', 'Read a book', 'Play games', 'Help in household','Is_South_India','Mean'])
NASdata_Mean_New1.head()

State  District  Gender  Age  Category  Same language  Siblings  Handicap  \
0      1         1       1    3         3              1         5         2   
1      1         1       2    3         4              2         5         2   
2      1         1       2    3         4              2         5         2   
4      1         1       2    3         3              2         5         2   
6      1         1       2    3         4              2         2         2   

   Father edu  Mother edu   ...    Conduct experiments  \
0           1           1   ...                      3   
1           2           1   ...                      3   
2           1           1   ...                      3   
4           1           1   ...                      3   
6           1           1   ...                      3   

   Solve science problems  Express science views  Watch TV  Read magazine  \
0                       3                      3         3              4   
1                       3                      3         4              4   
2                       3                      3         4              3   
4                       3                      3         2              3   
6                       3                      3         4              4   

   Read a book  Play games  Help in household  Is_South_India    Mean  
0            3           4                  4               1  24.075  
1            3           4                  4               1  25.570  
2            3           4                  4               1  27.780  
4            3           4                  4               1  14.550  
6            4           4                  4               1  40.740  

[5 rows x 61 columns]

In [10]:
NASdata_Mean_New1['State'].unique()

array([ 1,  2,  3,  4,  7,  9, 10, 12, 11, 14, 13, 15, 16, 20, 18, 19, 17,
       21, 22, 23, 24, 26, 27, 28, 29, 31, 30, 32,  0,  5, 25,  8,  6], dtype=int64)

At 95% Confidence Level,
Null Hypothesis H0: The students from South Indian states really excell in Maths and Science.
Alternate Hypothesis H1: The students from Other Indian states excell in Maths and Science than the students from South Indian states.

In [11]:
NASdata_Mean_South = NASdata_Mean_New1[NASdata_Mean_New1.Is_South_India==1]
NASdata_Mean_Others = NASdata_Mean_New1[NASdata_Mean_New1.Is_South_India==0]
NASdata_Mean_South['Mean'].describe()     #Mean value is 32.093647

count    41889.000000
mean        32.093647
std         13.463817
min          0.000000
25%         23.210000
50%         29.630000
75%         38.180000
max         96.430000
Name: Mean, dtype: float64

In [12]:
NASdata_Mean_Others['Mean'].describe()     #Mean value is 35.109863

count    106763.000000
mean         35.109863
std          16.039660
min           0.000000
25%          23.730000
50%          30.910000
75%          42.920000
max         100.000000
Name: Mean, dtype: float64

In [13]:
from scipy.stats import ttest_ind

In [14]:
#For the t-test calculation, take the mean values of the 'Mean' column grouped by each State for 'Is_South_India' == 1(South States) into Group 1 and 
#take the mean values of the 'Mean' column grouped by each State for 'Is_South_India' == 2(Other States) into Group 2
group1 = NASdata_Mean_New1[NASdata_Mean_New1.Is_South_India == 1].groupby('State')['Mean'].mean()
group2 = NASdata_Mean_New1[NASdata_Mean_New1.Is_South_India == 0].groupby('State')['Mean'].mean()

In [15]:
group1

State
0     35.475732
1     30.786039
9     36.496518
15    33.259847
16    33.968996
25    27.813491
28    29.743934
Name: Mean, dtype: float64

In [16]:
group2

State
2     31.979261
3     36.487588
4     30.931173
5     33.339750
6     43.777748
7     29.790347
8     42.105834
10    32.720812
11    30.809916
12    34.357438
13    37.022042
14    36.311994
17    30.515575
18    34.756615
19    39.838305
20    38.891621
21    35.211571
22    31.062119
23    36.303202
24    35.030092
26    33.853586
27    34.068671
29    42.504662
30    31.337616
31    43.802481
32    37.204475
Name: Mean, dtype: float64

In [17]:
#2 sample t test
t, p = ttest_ind(group1, group2)
print t     #t value
print p     #p value

-1.77862853078
0.0851101160725


Since the above p value(0.0851101160725) is much higher than 0.05 (95% Confidence Level), we have failed to reject the Null Hypothesis H0 and we can say that the students from Other Indian states(35.109863) excell in Maths and Science than the students from South Indian states(32.093647).